index.js

```diff
@@ -149,53 +195,74 @@ const parseAsType = {
 	date: (/* schema */) => ({ type: 'string', format: 'date-time' }),
 	boolean: (/* schema */) => ({ type: 'boolean' }),
 	alternatives: (schema, existingComponents, newComponentsByRef) => {
-		const index = meta(schema, 'swaggerIndex') || 0;
-
-		const matches = get(schema, [ '$_terms', 'matches' ]);
-		const firstItem = get(matches, [ 0 ]);
-
-		let itemsSchema;
-		if (firstItem.ref) {
-			if (schema._baseType && !firstItem.otherwise) {
-				itemsSchema = index ? firstItem.then : schema._baseType;
+		const matches = get(schema, '$_terms.matches');
+		const mode = `${get(schema, '_flags.match') || 'any'}Of`;
+
+		const alternatives = [];
+		for (const m of matches) {
+			if (m.ref) {
```

Comment:
@kroupacz  Can you give me a hint when m.ref will be set?

Reply:
@Mairu It will be set if you use "alternatives.conditional()" https://github.com/hapijs/joi/blob/master/API.md#alternativesconditionalcondition-options